ls1intum/Artemis
# `Assessment`: Fix number of assessments shown in course management statistics

### Checklist
#### General
#### Server

### Motivation and Context
@Santia-go noticed that the course cards on the course management side show a too high number of submissions. Students who just started the exercise but did not submitt anything still show up here, even though these participations don't have anything to assess. When moving to the assessment dashboard, a lower number is shown here (as expected).

### Description
I looked into the calls to `/for-assessment-dashboard` and `/stats-for-management-overview`. Both used different method of the submission repository, with `countByExerciseIdSubmittedBeforeDueDate()` actually filtering out participations without any submissions.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 3 Students
- 1 Exercise [of each type], manual assessment enabled and an assessment due date in the future

1. Both students start the exercise, but only one actually submits a solution
2. As an instructor, navigate to the course management page after the due date.
3. Make sure that the progress bar displaying the assessment status has a maxium of 1 submission to grade.
4. After the due date, add another (late) submission as the third student.
5. Make sure that the progress bar stil displays only one submission to assess.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
unchanged

### Screenshots
old:
![grafik](https://user-images.githubusercontent.com/26540346/162518389-c55ebcec-177b-4121-8bb4-319087c90177.png)

new: 
![grafik](https://user-images.githubusercontent.com/26540346/162518063-fb8a910c-73a7-4b14-ae7e-1a7ee8cef16f.png)

Assessment Dashboard (unchanged):
![grafik](https://user-images.githubusercontent.com/26540346/162518113-b06ec827-048f-4f84-937b-8a56369d0f6b.png)
